### PR TITLE
Support for new GO operator installation

### PR DESCRIPTION
**Pull request by mmajerni**
Signed-off-by: mmajerni <mmajerni@redhat.com>

### CODE REVIEW

**Comment by mmajerni** [comment]
@mcada  ^^

**Comment by mcada** [comment]
@tplevko please review the PR

File: src/test/java/apicurito/tests/TestRunner.java [hunk 1]
```diff
@@ -25,7 +26,11 @@
     public static void beforeTests() {
 
         if (OpenShiftUtils.xtf().getProject(TestConfiguration.openShiftNamespace()) == null) {
-            OpenShiftUtils.xtf().createProjectRequest(TestConfiguration.openShiftNamespace());
+            log.info("Creating new project " + TestConfiguration.openShiftNamespace());
```
Comment (tplevko) [inline src/test/java/apicurito/tests/TestRunner.java]: What's the reason behind abandoning the `createProjectRequest` method?

File: src/test/java/apicurito/tests/configuration/Component.java [hunk 2]
```diff
@@ -5,8 +5,9 @@
 @Getter
 public enum Component {
 
-    GENERATOR("fuse-apicurito-generator"),
-    UI("apicurito-ui");
+    UI("apicurito-ui"),
```
Comment (tplevko) [inline src/test/java/apicurito/tests/configuration/Component.java]: The generator is not deployed in the test suite?

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java [hunk 3]
```diff
@@ -62,38 +66,110 @@ public static void deployUsingTemplate() {
...
+    private static void deployRoleBinding() {
+        log.info("Deploying Role binding from " + TestConfiguration.apicuritoOperatorRoleBindingUrl());
+        final String output = OpenShiftUtils.binary().execute(
+                "create",
+                "-n", TestConfiguration.openShiftNamespace(),
+                "-f", TestConfiguration.apicuritoOperatorRoleBindingUrl()
+        );
+    }
+
+    private static void deployOperator() {
```
Comment (tplevko) [inline src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: These 6 methods seem to be quite repetitive, could these be squashed?

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java [hunk 4]
```diff
@@ -62,38 +66,110 @@ public static void deployUsingTemplate() {
...
+            //OpenShiftUtils.getInstance().extensions().deployments().inNamespace(TestConfiguration.openShiftNamespace()).delete();
+            OpenShiftUtils.getInstance().roles().delete();
+            OpenShiftUtils.getInstance().roleBindings().delete();
+            OpenShiftUtils.getInstance().customResourceDefinitions().delete();
         } catch (KubernetesClientException ex) {
             // Probably user does not have permissions to delete.. a nice exception will be printed when deploying
         }
-        OpenShiftUtils.getInstance().cleanAndAssert();
+        OpenShiftUtils.getInstance().clean();
+        OpenShiftUtils.getInstance().waiters().isProjectClean().waitFor();
```
Comment (tplevko) [inline src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: I'm having difficulties there - the previous steps don't seem to remove the operator deployments on my side.

File: pom.xml [hunk 5]
```diff
@@ -60,20 +68,12 @@
             <version>${logback.classic.version}</version>
         </dependency>
 
+
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply pom.xml]: deleted

File: src/test/java/apicurito/tests/TestRunner.java [hunk 6]
```diff
@@ -25,7 +26,11 @@
...
+            final String output = OpenShiftUtils.binary().execute(
+                    "new-project", TestConfiguration.openShiftNamespace()
+
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/TestRunner.java]: fixed

File: src/test/java/apicurito/tests/configuration/Component.java (hunk 2 above)
Comment (mmajerni) [reply src/test/java/apicurito/tests/configuration/Component.java]: template installation - yes 
operator installation - no 
but anyway we do not test generator and it is not needed to test apicurito

File: src/test/java/apicurito/tests/TestRunner.java (hunk 1 above)
Comment (mmajerni) [reply src/test/java/apicurito/tests/TestRunner.java]: ocp 4.1
createProjectRequest not working with ocp 4.1 right now.

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java (hunk 3 above)
Comment (mmajerni) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: 5 squashed, CR will be reworked in next PR

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java (hunk 4 above)
Comment (mmajerni) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: fixed

**Comment by mmajerni** [comment]
@tplevko added fixes ^^

File: src/test/java/apicurito/tests/TestRunner.java (hunk 1 above)
Comment (tplevko) [reply src/test/java/apicurito/tests/TestRunner.java]: Cool, didn't know that, thanks!

File: src/test/java/apicurito/tests/configuration/Component.java (hunk 2 above)
Comment (tplevko) [reply src/test/java/apicurito/tests/configuration/Component.java]: I see, thanks

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java (hunk 3 above)
Comment (tplevko) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: looks better, thanks

**Pull request merged**